lib/Dialect/TTIR/Transforms/Passes.cpp

```diff
@@ -511,6 +511,19 @@ getLegalTensorMemoryLayout(OperandConstraint operandConstraint,
   return TensorMemoryLayout::None;
 }
 
+inline Location appendInputSuffix(Location loc, int64_t operandIndex) {
+  if (isa<NameLoc>(loc)) {
+    NameLoc oldLoc = mlir::cast<NameLoc>(loc);
+    StringAttr newName = StringAttr::get(
+        loc->getContext(), oldLoc.getName().str() + "_in_" +
+                               std::to_string(operandIndex) + "_layout");
+
+    return NameLoc::get(newName, oldLoc.getChildLoc());
+  }
+
+  return loc;
+}
```

Comment:
Just for my own understanding, this preserves the original file location right?  It results in kind of that nested loc syntax?

Reply:
Yes, we use FileLineColLoc nested inside a NameLoc. This preserves the same FileLineColLoc with a new name.